File: javascript/Shubham Sigdar/Fake binary.js
// Given a string of digits, you should replace any digit below 5 with '0' and any digit 5 and above with '1'. Return the resulting string.

// Note: input will never be an empty string
//P:string of digit
//R:Return the resulting string.
//E:
//P:split string into array
//map
//condition
//replace
function fakeBin(x) {
    return x.split('').map(n => n < 5 ? 0 : 1).join('');
}


console.log(fakeBin('45385593107843568'), '01011110001100111')
console.log(fakeBin('509321967506747'), '101000111101101')
console.log(fakeBin('366058562030849490134388085'), '011011110000101010000011011')
// fakeBin('4538')

File: java/Shubham Sigdar/FakeBinary.java
//Given a string of digits, you should replace any digit below 5 with '0' and any digit 5 and above with '1'.
// Return the resulting string.
//
//        Note: input will never be an empty string
public class FakeBinary {
    public static String fakeBin(String numberString) {
//       String[] str = numberString.split("");
        String res="";
        for(int i=0;i<numberString.length();i++){
            if(numberString.charAt(i)<'5'){
                res+="0";
            }else{
                res+="1";
            }
        }
        return res;
    }
//    final char c[] = numberString.toCharArray();
//        for (int i = 0; i < c.length; i++)
//    c[i] = c[i] < '5' ? '0' : '1';
//        return new String(c);
}
